src/solvers/petsc_nonlinear_solver.C

```diff
@@ -181,7 +181,8 @@ extern "C"
     else
       libmesh_error_msg("Error! Unable to compute residual and/or Jacobian!");
 
-    rc.sys.get_dof_map().enforce_constraints_on_residual(rc.sys, &R, rc.sys.current_local_solution.get());
+    PetscVector<Number> X(x, rc.sys.comm());
```

Comment:
Wait.  @lindsayad are we now making a copy of this... and then killing it each iteration?

Reply:
This does not copy the PETSc Vec object. The doxygen for this constructor is [here](https://libmesh.github.io/doxygen/classlibMesh_1_1PetscVector.html#a237268dc452928c69bb5b8485e0f4d7b)

Reply:
right - I see now.